clang/include/clang/CConv/ProgramVar.h

```diff
@@ -355,11 +346,25 @@ class ProgramVar {
   std::string verboseStr();
   ProgramVar *makeCopy(BoundsKey NK);
   virtual ~ProgramVar() { }
+
+  static ProgramVar *createNewProgramVar(BoundsKey VK, std::string VName,
+                                         const ProgramVarScope *PVS,
+                                         bool IsCons = false);
 private:
   BoundsKey K;
   std::string VarName;
   const ProgramVarScope *VScope;
   bool IsConstant;
+  static std::set<ProgramVar *> AllProgramVars;
```

Comment:
So you are just keeping around a set of these things so they are not freed prematurely? Is this going to have the opposite problem of being a memory leak?

Reply:
Nope. We only free a `ProgramVar` during a brain transplant. However, similar to `ConstraintVariables`, we cannot always be sure that some `ProgramVar` is never used. We can use a BiMap to handle this, but, I think this is a more robust and efficient solution with not-so-high memory overhead.

Reply:
I guess it might be useful to see if there's a way to tell when it's not needed, rather than just give up. Maybe leave a comment in the code that says that we should only be freeing `ProgramVar` during `brainTransplant` but we aren't sure when it's safe to do so. It's a TODO to figure that out, if it's possible.